### PR TITLE
[2.5] Add authentication to Client API messages

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

### Description

Messages between CJ and Client API Agent could be relayed through the Server. Without proper authentication headers, these messages will be blocked. This PR adds auth headers to all such messages.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/utils/fed_utils.py [hunk 1]
```diff
@@ -415,20 +414,15 @@ def set_scope_prop(scope_name: str, key: str, value: Any):
...
         scope_name: name of the scope
         key: key of the scope
+        default: value to return if the prop is not found
     Returns:
     """
-    check_str("scope_name", scope_name)
-    check_str("key", key)
-    data_bus = DataBus()
-    return data_bus.get_data(_scope_prop_key(scope_name, key))
+    return get_scope_property(scope_name, key, default)
```
Comment (SYangster) [inline nvflare/private/fed/utils/fed_utils.py:428]: What is the reasoning for having these functions in addition to the ones added in fuel.data_event.utils?

**Review by SYangster — COMMENTED** [review]
Good fix, minor comment

File: nvflare/private/fed/utils/fed_utils.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/private/fed/utils/fed_utils.py:428]: Just for backward compatibility - some places are already using fed_utils.

The real reason to move this function to nvflare.fuel.data_event.utils is for cleaner dependency. We don't want stuff in app_common (e.g. those client API stuff) to depend on fed_utils, which is in "private".

**Review by SYangster — APPROVED** [review]

**Pull request merged**